src/test/groovy/hudson/plugins/gradle/enriched/ScanDetailServiceDefaultImplTest.groovy

```diff
@@ -0,0 +1,212 @@
+        scanDetailService.httpClientFactory = httpClientFactory
+        def httpClient = Stub(CloseableHttpClient)
+        def response1 = Stub(CloseableHttpResponse)
+        response1.getStatusLine() >> new BasicStatusLine(HttpVersion.HTTP_1_1, HttpStatus.SC_OK, "")
+        response1.getEntity() >> new StringEntity(
+                """
+                    {
+                        "buildToolType": "${buildToolType}",
+                        "buildToolVersion": "7.5.1"   
+                    }
+                """.stripIndent())
+        def response2 = Stub(CloseableHttpResponse)
+        response2.getStatusLine() >> new BasicStatusLine(HttpVersion.HTTP_1_1, HttpStatus.SC_OK, "")
+        response2.getEntity() >> new StringEntity("{This is not valid JSON}")
+        httpClientFactory.buildHttpClient(_,_,_) >> httpClient
+        httpClient.execute(_) >>> [response1, response2]
+
+        when:
+        def scanDetail = scanDetailService.getScanDetail("https://foo.bar/s/scanId")
+
+        then:
+        scanDetail == Optional.empty()
+
+        where:
+        buildToolType << ["gradle", "maven"]
+    }
+
+    @Unroll
+    def 'Get scan detail with unrecognized JSON field'() {
+        given:
+        def scanDetailService = new ScanDetailService(getTestConfig())
+        def httpClientFactory = Stub(HttpClientFactory)
+        scanDetailService.httpClientFactory = httpClientFactory
+        def httpClient = Stub(CloseableHttpClient)
+        def response1 = Stub(CloseableHttpResponse)
+        response1.getStatusLine() >> new BasicStatusLine(HttpVersion.HTTP_1_1, HttpStatus.SC_OK, "")
+        response1.getEntity() >> new StringEntity(
+                """
+                    {
+                        "buildToolType": "${buildToolType}",
+                        "buildToolVersion": "7.5.1"   
+                    }
+                """.stripIndent())
+        def response2 = Stub(CloseableHttpResponse)
+        response2.getStatusLine() >> new BasicStatusLine(HttpVersion.HTTP_1_1, HttpStatus.SC_OK, "")
+        response2.getEntity() >> new StringEntity(httpResponseBody)
+        httpClientFactory.buildHttpClient(_,_,_) >> httpClient
+        httpClient.execute(_) >>> [response1, response2]
+        httpClientFactory.buildHttpClient(_,_,_) >> httpClient
+
+        when:
+        def scanDetailResult = scanDetailService.getScanDetail("https://foo.bar/s/scanId")
+
+        then:
+        def scanDetail = scanDetailResult.get()
+        scanDetail.url == "https://foo.bar/s/scanId"
+        scanDetail.buildToolType.toString() == buildToolType.toUpperCase()
+        scanDetail.buildToolVersion == "7.5.1"
+        scanDetail.projectName == "project"
+        scanDetail.tasks == [ "clean", "build" ]
+        !scanDetail.hasFailed
+
+        where:
+        buildToolType | httpResponseBody
+        "gradle"      | "{\"foo\":\"bar\",\"rootProjectName\":\"project\",\"requestedTasks\":[\"clean\",\"build\"],\"hasFailed\":false}"
```

Comment:
It is possible to use `'` or `"""` quotes to define a string, then there is no need to escape `"`. It's a bit easier to read

```suggestion
        "gradle"      | '{"foo":"bar","rootProjectName":"project","requestedTasks":["clean","build"],"hasFailed":false}'
```